misk-crypto/src/test/kotlin/misk/crypto/CryptoModuleTest.kt

```diff
@@ -116,6 +109,37 @@ class CryptoModuleTest {
     }.isInstanceOf(GeneralSecurityException::class.java)
   }
 
+  @Test
+  fun testLogsWarningWithoutEnvelopeKey() {
+    val kh = KeysetHandle.generateNew(AeadKeyTemplates.AES256_CTR_HMAC_SHA256)
+    val encryptedKey = generateObsoleteEncryptedKey(kh)
+    val key = Key("name", KeyType.AEAD, encryptedKey)
+    val kr = KeyReader()
+    val client = FakeKmsClient()
+    val (out, _) = captureOutput {
+      kr.readKey(key, "aws-kms://some-uri", client)
+    }
+    assertThat(out).contains("WARN")
+    assertThat(out).contains("using obsolete key format")
+  }
+
+  // Run a block, capturing stdout and stderr; return as strings
+  private fun captureOutput(f: () -> Any?): Pair<String, String> {
```

Comment:
I'm not entirely sure about it, but I think Kotlin/Misk might have some tools for capturing/analyzing logs and output in tests

Reply:
Yeah, take a look at `misk.logging.LogCollector`